From 3d06f34aa89698f74e743b9ec023eafc19827cba Mon Sep 17 00:00:00 2001
Message-Id: <3d06f34aa89698f74e743b9ec023eafc19827cba.1657024365.git.geliang.tang@suse.com>
In-Reply-To: <3a7b35b899dedd29468301a3cbc4fa48a49e2131.1657024365.git.geliang.tang@suse.com>
References: <3a7b35b899dedd29468301a3cbc4fa48a49e2131.1657024365.git.geliang.tang@suse.com>
From: Song Liu <songliubraving@fb.com>
Date: Fri, 12 Feb 2021 10:31:06 -0800
Subject: [PATCH 2/3] bpf: Allow bpf_d_path in bpf_iter program
Git-commit: 3d06f34aa89698f74e743b9ec023eafc19827cba
Patch-mainline: v5.12-rc2
References: bsn#295
Modified-by-SEL: No

task_file and task_vma iter programs have access to file->f_path. Enable
bpf_d_path to print paths of these file.

Signed-off-by: Song Liu <songliubraving@fb.com>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Acked-by: Yonghong Song <yhs@fb.com>
Link: https://lore.kernel.org/bpf/20210212183107.50963-3-songliubraving@fb.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 kernel/trace/bpf_trace.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/trace/bpf_trace.c b/kernel/trace/bpf_trace.c
index 845b2168e006..0b9e4fd9c61b 100644
--- a/kernel/trace/bpf_trace.c
+++ b/kernel/trace/bpf_trace.c
@@ -1191,6 +1191,10 @@ BTF_SET_END(btf_allowlist_d_path)
 
 static bool bpf_d_path_allowed(const struct bpf_prog *prog)
 {
+	if (prog->type == BPF_PROG_TYPE_TRACING &&
+	    prog->expected_attach_type == BPF_TRACE_ITER)
+		return true;
+
 	if (prog->type == BPF_PROG_TYPE_LSM)
 		return bpf_lsm_is_sleepable_hook(prog->aux->attach_btf_id);
 
-- 
2.35.3

